be2-scala/src/test/scala/ch/epfl/pop/json/MessageDataProtocolSuite.scala

```diff
@@ -15,16 +15,16 @@ class MessageDataProtocolSuite extends FunSuite with Matchers {
     def shouldEqualTo(o: CastVoteElection): Unit = {
 
       @scala.annotation.tailrec
-      def checkVoteElectionLists(l1: List[VoteElection], l2: List[VoteElection]) {
-        l1.length should equal (l2.length)
+      def checkVoteElectionLists(l1: List[VoteElection], l2: List[VoteElection]): Unit = {
+        l1.length should equal(l2.length)
 
-        (l1, l2) match {
+        ((l1, l2): @unchecked) match {
```

Comment:
Do we need an `@unchecked` here? Is there no better way?

Reply:
The unchecked is to avoid the "non-exhaustive match warning" when I know the compiler is wrong. We can definitely modify it but the code will be very slightly less efficient/clean. However I think you are definitely right (showing a clean implementation to students > gaining a few clock cycles in the tests ^^)! I'll change it